### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by Darkclainer**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,10 +2,99 @@
 
 This is a code generator for Go that alleviates JSON marshaling/unmarshaling unrelated structs in typed fashion.
 
-In Go you can marshal/unmarshal concrete structures, but what if you need to marshal/unmarshal different unrelated structures in one place.
-In this case you can use empty `interface{}` or something more convoluted like `map[string]interface{}`, but in this case you will lose information and fight with type system (more like type system will ignore you). 
-This package partially solves this problem, with it you can select set of structures that can be marshaled/unmarshaled behind interface and preserve its type.
+Imagine, that you send or receive JSON object with some key `config`.
```
Comment (franchb) [inline README.md]:
```suggestion
Imagine, that you send or receive a JSON object with some key `config`. 
```

File: README.md [hunk 2]
```diff
@@ -2,10 +2,99 @@
 
 This is a code generator for Go that alleviates JSON marshaling/unmarshaling unrelated structs in typed fashion.
 
-In Go you can marshal/unmarshal concrete structures, but what if you need to marshal/unmarshal different unrelated structures in one place.
-In this case you can use empty `interface{}` or something more convoluted like `map[string]interface{}`, but in this case you will lose information and fight with type system (more like type system will ignore you). 
-This package partially solves this problem, with it you can select set of structures that can be marshaled/unmarshaled behind interface and preserve its type.
+Imagine, that you send or receive JSON object with some key `config`. 
+The value of this field can correspond to two structs of your Go program: `FooConfig` and `BarConfig`.
+So, the field `Config` in your struct must be able to hold value of two possible types.
```
Comment (franchb) [inline README.md]:
```suggestion
So, the field `Config` in your struct must be able to hold a value of two possible types.
```

File: README.md [hunk 3]
```diff
@@ -2,10 +2,99 @@
...
+In this case you basically have next options:
```
Comment (franchb) [inline README.md]:
```suggestion
In this case, you have the following options:
```

File: README.md [hunk 4]
```diff
@@ -2,10 +2,99 @@
...
-In Go you can marshal/unmarshal concrete structures, but what if you need to marshal/unmarshal different unrelated structures in one place.
-In this case you can use empty `interface{}` or something more convoluted like `map[string]interface{}`, but in this case you will lose information and fight with type system (more like type system will ignore you). 
-This package partially solves this problem, with it you can select set of structures that can be marshaled/unmarshaled behind interface and preserve its type.
+Imagine, that you send or receive JSON object with some key `config`. 
+The value of this field can correspond to two structs of your Go program: `FooConfig` and `BarConfig`.
+So, the field `Config` in your struct must be able to hold value of two possible types.
+In this case you basically have next options: 
 
-## Details
+1. You can declare field `Config` as `interface{}` and somehow determine what type you should expect, assign object of this type to `Config`
```
Comment (franchb) [inline README.md]:
```suggestion
1. You can declare field `Config` as `interface{}` and somehow determine what type you should expect, assign an object of this type to `Config` 
```

File: README.md [hunk 5]
```diff
@@ -2,10 +2,99 @@
...
+Imagine, that you send or receive JSON object with some key `config`. 
+The value of this field can correspond to two structs of your Go program: `FooConfig` and `BarConfig`.
+So, the field `Config` in your struct must be able to hold value of two possible types.
+In this case you basically have next options: 
 
-## Details
+1. You can declare field `Config` as `interface{}` and somehow determine what type you should expect, assign object of this type to `Config` 
+and then unmarshal object.
+1. You can unmarshal field `Config` separately.
+1. You can implement custom `MarshalJSON`/`UnmarshalJSON` for third type that automatically will handle this cases.
```
Comment (franchb) [inline README.md]:
```suggestion
1. You can implement custom `MarshalJSON`/`UnmarshalJSON` for the third type that automatically will handle these cases.
```

File: README.md [hunk 6]
```diff
@@ -2,10 +2,99 @@
...
+
+This package provides means to generate all boilerplate code for third case.
```
Comment (franchb) [inline README.md]:
```suggestion
This package provides means to generate all boilerplate code for the third case.
```

File: README.md [hunk 7]
```diff
@@ -2,10 +2,99 @@
...
+
+	Output path where generated code should be saved.
+
+* `-package` string
+
+	Package name in generated file (default to GOPACKAGE).
+
+* `-typed` string
+
+	Name of struct that will used for typed interface (default to %%interface%%Typed.
```
Comment (franchb) [inline README.md]:
```suggestion
	Name of the struct that will be used for typed interface (default to `%%interface%%Typed`).
```

File: README.md [hunk 8]
```diff
@@ -2,10 +2,99 @@
...
+
+Each name in position argument should be name of struct.
```
Comment (franchb) [inline README.md]:
```suggestion
Each name in position argument should be the name of the struct. 
```

File: README.md [hunk 9]
```diff
@@ -2,10 +2,99 @@
...
+* `-package` string
+
+	Package name in generated file (default to GOPACKAGE).
+
+* `-typed` string
+
+	Name of struct that will used for typed interface (default to %%interface%%Typed.
+
+Each name in position argument should be name of struct. 
+You can set alias for struct name like this: `foo=*FooConfig`.
```
Comment (franchb) [inline README.md]:
```suggestion
You can set an alias for struct name like this: `foo=*FooConfig`.
```

File: README.md [hunk 10]
```diff
@@ -2,10 +2,99 @@
...
+type FooConfig struct {
+	Foo int
+}
+
+type BarConfig struct {
+	Bar string
+}
+```
+
+Then you must declare interface that will hold either of these structs.
```
Comment (franchb) [inline README.md]:
```suggestion
Then you must declare an interface that will hold either of these structs.
```

File: README.md [hunk 11]
```diff
@@ -2,10 +2,99 @@
...
+The interface must have method `TypedJSON` with special signature.
```
Comment (franchb) [inline README.md]:
```suggestion
The interface must have the method `TypedJSON` with a special signature. 
```

File: README.md [hunk 12]
```diff
@@ -2,10 +2,99 @@
...
+}
+
+type BarConfig struct {
+	Bar string
+}
+```
+
+Then you must declare interface that will hold either of these structs.
+The interface must have method `TypedJSON` with special signature. 
+This method help compiler to work with types.
```
Comment (franchb) [inline README.md]:
```suggestion
This method will advise compiler to work with types.
```

**Review by franchb — CHANGES_REQUESTED** [review]
Minor grammatical suggestions introduced.

File: README.md [hunk 13]
```diff
@@ -2,10 +2,98 @@
...
+
+	Output path where generated code should be saved.
+
+* `-package` string
+
+	Package name in generated file (default to GOPACKAGE).
+
+* `-typed` string
+
+	Name of struct that will used for typed interface (default to %%interface%%Typed.
```
Comment (Darkclainer) [inline README.md]:
```suggestion
	The name of the struct that will be used for typed interface (default to `{{interface}}{{Typed}}`).
```

**Review by franchb — APPROVED** [review]

**Pull request merged**